Writing-state-migrations.md

```diff
@@ -2,73 +2,26 @@ If your PR changes the properties of an exploration or state (or other structure
-     - core/templates/pages/exploration-editor-page/editor-tab/training-panel/training-panel.component.spec.ts
-     - core/templates/pages/exploration-editor-page/services/exploration-states.service.spec.ts
-     - core/templates/pages/exploration-editor-page/services/graph-data.service.spec.ts
-     - core/templates/pages/exploration-editor-page/services/parameter-metadata.service.spec.ts
-     - core/templates/pages/exploration-editor-page/services/router.service.spec.ts
-     - core/templates/pages/exploration-editor-page/statistics-tab/statistics-tab.component.spec.ts
-     - core/templates/pages/exploration-editor-page/translation-tab/services/translation-status.service.spec.ts
-     - core/templates/pages/exploration-editor-page/translation-tab/state-translation-editor/state-translation-editor.component.spec.ts
-     - core/templates/pages/exploration-player-page/services/answer-classification.service.spec.ts
-     - core/templates/pages/exploration-player-page/services/audio-preloader.service.spec.ts
-     - core/templates/pages/exploration-player-page/services/extract-image-filenames-from-state.service.spec.ts
-     - core/templates/pages/exploration-player-page/services/image-preloader.service.spec.ts
-     - core/templates/pages/exploration-player-page/services/learner-answer-info.service.spec.ts
-     - core/templates/pages/skill-editor-page/services/question-creation-service.spec.ts
-     - core/templates/pages/skill-editor-page/skill-preview-tab/skill-preview-tab.controller.spec.ts
-     - core/templates/services/question-validation.service.spec.ts
-     - core/templates/services/state-interaction-stats.service.spec.ts
-     - core/templates/services/state-top-answers-stats.service.spec.ts
+2. Increment the `CURRENT_STATE_SCHEMA_VERSION` in the _feconf.py_ file.
+3. Increment the `CURRENT_EXP_SCHEMA_VERSION` in the _exp_domain.py_ file and similar changes in the _question_domain.py_ file.
+4. Start with writing `_convert_states_v{old_state_version}_dict_to_v{old_state_version + 1}_dict` method in _exp_domain.py_ files under `Exploration` class and in _question_domain.py_ under `Question` class. In _exp_domain.py_, update the `Exploration._migrate_to_latest_yaml_version` method to use your conversion function (not required for _question_domain.py_, this is done automatically).
+5. Write a conversion function `_convert_states_v{old_state_version}_dict_to_v{old_state_version + 1}_dict` in _draft_upgrade_services.py_ that makes appropriate upgrades to data that resides in `ExplorationChange` lists. Here you have to check if the changes that you do in your conversion function affect the `draft_change_list`, if yes, then make sure to update the drafts or raise `InvalidDraftConversionException` which will drop the changes in the drafts (this needs to be done in cases where changes to drafts would be too complicated). If no changes need to be done to drafts you can simply return the `draft_change_list`.
```

Comment:
"that you do in your conversion function" → please explain which conversion function exactly

Maybe split this text into multiple subpoints for better readability

Reply:
Done